Server/ConcordiaCurriculumManager/Repositories/GroupRepository.cs

```diff
@@ -61,4 +63,42 @@ public async Task<bool> RemoveUserFromGroup(Guid userId, Guid groupId)
         }
         return false;
     }
+
+    public async Task<bool> AddGroupMaster(Guid userId, Guid groupId)
+    {
+        var group = await _dbContext.Groups
+                                    .Include(g => g.Members)
+                                    .Include(g => g.GroupMasters)
+                                    .FirstOrDefaultAsync(g => g.Id == groupId);
+
+        if (group == null)
+        {
+            return false;
+        }
+
+        var member = group.Members.FirstOrDefault(u => u.Id == userId);
+        if (member == null)
+        {
+            return false;
+        }
+
+        var isAlreadyGroupMaster = group.GroupMasters.Any(u => u.Id == userId);
+        if (isAlreadyGroupMaster)
+        {
+            return false;
+        }
+
+        group.GroupMasters.Add(member);
+        return await _dbContext.SaveChangesAsync() > 0;
+    }
+
+    public async Task<Group> GetGroupWithMasters(Guid groupId)
```

Comment:
I don't think this method needs to exist, instead you can simply use the existing GetGroupById method. To get this working, just modify the GroupSelector method in the ObjectsSelectors to also select GroupMasters. It should look like so:

```
public static Expression<Func<Group, Group>> GroupSelector() => group => new Group
{
    Id = group.Id,
    Name = group.Name,
    Members = (List<User>)group.Members.Select(user => new User
    {
        Id = user.Id,
        FirstName = user.FirstName,
        LastName = user.LastName,
        Email = user.Email,
        Password = user.Password,
        Roles = (List<Role>)user.Roles.Select(role => new Role
        {
            UserRole = role.UserRole
        })
    }),
    GroupMasters = (List<User>)group.GroupMasters.Select(user => new User
    {
        Id = user.Id,
        FirstName = user.FirstName,
        LastName = user.LastName,
        Email = user.Email,
        Password = user.Password,
        Roles = (List<Role>)user.Roles.Select(role => new Role
        {
            UserRole = role.UserRole
        })
    })
};
```

In addition, in GroupDTO, you can also create a GroupMasters field like so:
```
public class GroupDTO
{
    public required Guid Id { get; set; }
    public required string Name { get; set; }
    public required List<UserDTO> Members { get; set; } = new();
    public required List<UserDTO> GroupMasters { get; set; } = new();
}
```
So that the GetGroupById and GetAllGroups endpoints return the group masters (which is behaviour we want)